### DISCUSSION TITLE
Support Rust 1.47.0 for Ubuntu

### DISCUSSION THREAD

**Discussion by gaborbernat** [brettcannon/python-launcher#96] [Ideas]
**Describe the bug**
Fails to compile on docker image `ubuntu:latest`.

**To Reproduce**
```bash
cargo install python-launcher

   Compiling python-launcher v0.15.0
error[E0658]: use of unstable library feature 'bool_to_option'
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/python-launcher-0.15.0/src/cli.rs:164:29
    |
164 |         venv_path.is_file().then(|| venv_path)
    |                             ^^^^
    |
    = note: see issue #64260 <https://github.com/rust-lang/rust/issues/64260> for more information
error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
error: could not compile `python-launcher`.
```

**Expected behaviour**
The build to succeed. 

**System Details (please complete the following information):**
 - OS: ubuntu:latest
 - Shell: bash
 - Launcher Version: latest

rust (1.47) & cargo (0.47) installed via `apt-get`. Note it works if one installs the latest rust via https://rustup.rs.

**Comment by brettcannon** [comment]
I have only worried about compatibility with stable (https://github.com/brettcannon/python-launcher#how-do-i-install-the-launcher), so I'm considering this a feature request to backport to [Rust 1.47.0](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1470-2020-10-08).

The specific API in question is [bool::then()](https://doc.rust-lang.org/stable/std/primitive.bool.html#method.then) which went stable in Rust 1.50.

**Reply by gaborbernat** [reply]
Would be nice if cargo would fail here that rust 1.50 or later is required 🤔
